install.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  ErrorKind,
+  Process,
+  ProcessStatus,
+  args,
+  chmod,
+  copyFile,
+  env,
+  exit,
+  lstat,
+  platform,
+  symlink,
+  writeFile,
+  removeAll,
+  rename,
+  run,
+  makeTempDir
+} from "deno";
+
+import { mkdirp } from "https://deno.land/x/std/mkdirp/mkdirp.ts";
+import { join, win32 } from "https://deno.land/x/path/index.ts";
+
+const WIN32: boolean = platform.os === "win";
+
+const proc_env: { [key: string]: any } = env();
+
+function get_home(): string {
+  return WIN32 ? win32.resolve("C:", proc_env.HOMEPATH) : proc_env.HOME;
+}
+
+const DENO_REPO_URL: string = "https://github.com/denoland/deno";
+const LATEST_RELEASE_URL: string = `${DENO_REPO_URL}/releases/latest`;
+const TAG_RELEASE_URL: string = `${DENO_REPO_URL}/releases/tag`;
+
+const DENO_DIR: string = join(get_home(), ".deno");
+const DENO_BIN_DIR: string = join(DENO_DIR, "bin");
+const DENO_BIN: string = join(DENO_BIN_DIR, WIN32 ? "deno.exe" : "deno");
+const OLD_DENO_BIN: string = DENO_BIN.replace(/deno(\.exe)?$/, "old_deno$1");
+const DENO_LINK: string = join(
+  WIN32 ? win32.resolve("C:", "Windows", "System32") : "/usr/local/bin",
```

Comment:
You could use the `SystemRoot` environment variable here (e.g. `SystemRoot=C:\WINDOWS`).

```suggestion
  WIN32 ? win32.resolve(proc_env.SystemRoot, "System32") : "/usr/local/bin",
```